src/ORM/Hierarchy/Hierarchy.php

```diff
@@ -360,6 +381,127 @@ public function AllChildrenIncludingDeleted()
         return $stageChildren;
     }
 
+    /**
+     * Return direct children in the hierarchy
+     *
+     * This does much the same result as AllChildrenIncludingDeleted() and is far more performant,
+     * though there are some key differences:
+     * - Results will be cached in-memory to save on the number of database queries required
+     * - Database queries will be larger `WHERE "ID" IN (?, ?) style queries rather than many `WHERE "ID" = ?
+     * - While Hierarchy.node_threshold_total has not been exceeded, it will cache descendant record as
+     *   well to save future queries
+     * - Does not included deleted records, which could be done via Versioned::updateListToAlsoIncludeDeleted($children)
+     */
+    public function getAllChildren(): ArrayList
```

Comment:
```suggestion
    public function getAllChildren(): SSList
```
I assume there's nothing inherent to `ArrayList` in particular we need where this is returned, so using `SSList` as the return type allows us to change it e.g. if we figure out a way to performantly use `DataList` in the future.

Reply:
The old method we're (very debatably, this has increased the complexity here a fair bit for very little benefit IMO) keeping `AllChildrenIncludingDeleted` returns an ArrayList, so this method probably should too.

I've also added an extension hook here where an ArrayList is what's expected (though it's not strictly typed, it's effectively still API), though I'm OK with ditching that extension hook if need be. I only really added it because `AllChildrenIncludingDeleted` has an extension hook

Note I've deprecated the old method (remove in CMS 7), there's really no reason to use it.

Reply:
I don't understand your reasoning but this isn't worth arguing about. Leave as is.